app/models/host/managed.rb

```
@@ -786,6 +787,46 @@ def validate_media?
     managed && pxe_build? && build?
   end
 
+  def template_used(provisioning = nil)
+    kinds = if provisioning == 'image'
+              cr     = ComputeResource.find_by_id(self.compute_resource_id)
+              images = cr.try(:images)
+              if images.nil?
+                [TemplateKind.find_by_name('finish')]
+              else
+                uuid       = self.compute_attributes.cr.image_param_name
+                image_kind = images.find_by_uuid(uuid).try(:user_data) ? 'user_data' : 'finish'
+                [TemplateKind.find_by_name(image_kind)]
+              end
+            else
+              TemplateKind.all
+            end
+
+    templates = kinds.map do |kind|
+      ConfigTemplate.find_template({:kind               => kind.name,
+                                    :operatingsystem_id => self.operatingsystem_id,
+                                    :hostgroup_id       => self.hostgroup_id,
+                                    :environment_id     => self.environment_id
+                                   })
+    end.compact
+  end
+
+  def host_templates_valid?
+    return false if self.template_used.empty?
+    self.template_used.each do | template |
+      begin
+        # Set instance variable for unattended_render function
+        @host = self
+        valid_template = unattended_render(template.template)
+        return false if valid_template.blank?
+      rescue
```

Try to rescue the rendering error you are expecting, this will catch all exceptions and we might miss some valuable stack traces.